### ISSUE TITLE
Should we have a Code of Conduct?

### ISSUE THREAD

**Issue by williammartin** [storyscript/community#3]
### Topic

Code of Conducts are important tools in managing communities. Should we have one, and if so is there an existing one that suits us, or do we need to write our own?

### Notes

The [Open Source Guide](https://opensource.guide/code-of-conduct/) describes well the purpose of a Code of Conduct.

**Comment by jayvdb** [comment]
Yes, we should have one, and atm we dont have the bandwidth to be rolling out own.  IMO we take an existing one and use it as our first one, and then look for another one when we outgrow the first one.

**Comment by williammartin** [comment]
Let's fine examples of Codes of Conduct we think are appropriate for Storyscript. I'll assign this to myself.

**Comment by jayvdb** [comment]
Obligatory https://coala.io/coc , as that was the result of a fairly broad review of existing ones , at the time, 3-ish years ago.  We took the most applicable, and slightly tweaked it.

Note that we can create a `.github` *repository* for these files, and they will appear in all repos.  https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization

But the files are meaningless unless there is additional & appropriate processes in place to receive and investigate reports, and enforce the CoC.

**Comment by jayvdb** [comment]
https://foundation.wikimedia.org/wiki/Code_of_conduct_policy is another I was involved in developing, prior to coala, but I havent been involved since it was enacted so dont know how well it works in practise - except that I know it has had a few community-splitting cases to resolve.  The good thing about this one is there is an ongoing public discussion about it at https://www.mediawiki.org/wiki/Talk:Code_of_Conduct, and it is widely translated.  https://www.contributor-covenant.org/translations also has quite a few translations.  IMO this is a very important selection criteria, and a good reason to not create a custom one.  The process of translating often highlights English-centrism in the text which needs to be fixed, as that causes the CoC to be a form of discrimination against ESL.

**Comment by williammartin** [comment]
It sounds like you have a much better idea of where to start in this regard than me. If you think this is a good shortlist, I could focus my efforts on these and give me feedback?

**Comment by williammartin** [comment]
@jayvdb I had a read through these, thank you for sharing. I think that the mediawiki CoC perhaps has a bit too much emphasis on events for us to lift and shift right now. I like both the Coala and Contributor Covenants and am in agreement about being inclusive for ESL contributors.

With that in mind, do you think the default Contribuor Covenant is acceptable or are there any tweaks you would want to make?

**Comment by jayvdb** [comment]
I've re-read it and I think we should go with Contributor Covenant as-is , replacing the "project team" email address with one or more people.  We can look to customise or choose another one later, when the cost/benefit warrants it.

**Comment by williammartin** [comment]
I guess as follows:

---

### Our Pledge

In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to make participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.

### Our Standards

Examples of behavior that contributes to creating a positive environment include:

* Using welcoming and inclusive language
* Being respectful of differing viewpoints and experiences
* Gracefully accepting constructive criticism
* Focusing on what is best for the community
* Showing empathy towards other community members

Examples of unacceptable behavior by participants include:

* The use of sexualized language or imagery and unwelcome sexual attention or advances
* Trolling, insulting/derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others’ private information, such as a physical or electronic address, without explicit permission
* Other conduct which could reasonably be considered inappropriate in a professional setting
Our Responsibilities

Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.

Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

### Scope

This Code of Conduct applies within all project spaces, and it also applies when an individual is representing the project or its community in public spaces. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.

### Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at conduct@storyscript.io. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.

Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project’s leadership.

### Attribution

This Code of Conduct is adapted from the Contributor Covenant, version 1.4, available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html

For answers to common questions about this code of conduct, see https://www.contributor-covenant.org/faq

--- 

Action items would be to get this reviewed perhaps by legal, by others of the team (who?) so we have quorum, create the email address, and add the CoC to the necessary location if approved? Anything else that comes to mind?

**Comment by stevepeak** [comment]
@williammartin great work! I can setup email, and be responsible for it for the time being. Can we push this into into repositories/readme's? 👍

**Comment by wilzbach** [comment]
> Can we push this into into repositories/readme's

It's already set organization-wide:
- https://github.com/microservices/.github/blob/master/CODE_OF_CONDUCT.md
- https://github.com/storyscript/.github/blob/master/CODE_OF_CONDUCT.md

So there are only two places which require an update once this email has been setup.

Though, of course, the individual repositories could have a reference like this shield:

[![Contributor
Covenant](https://img.shields.io/badge/Contributor%20Covenant-v1.4%20adopted-ff69b4.svg)](https://github.com/storyscript/.github/blob/master/CODE_OF_CONDUCT.md)
